### PR TITLE
negar/callpute_only_show_if_dur

### PR DESCRIPTION
**Pull request by negarn**
only show checkbox if callputequals is allowed for current duration

### CODE REVIEW

File: src/javascript/app/pages/trade/process.js [hunk 1]
```diff
@@ -217,9 +219,22 @@ const Process = (() => {
...
-    const displayEquals = (formname = Defaults.get('formname'), contracts = State.getResponse('contracts_for.available')) => {
-        const el_equals = getElementById('callputequal');
-        if (/^(callputequal|risefall)$/.test(formname) && (contracts || []).find(contract => contract.contract_category === 'callputequal')) {
+    const duration_type = {
+        t: 'tick',
+        s: 'intraday',
+        m: 'intraday',
+        h: 'intraday',
+        d: 'daily',
+    };
```
Comment (OskarAhl) [inline src/javascript/app/pages/trade/process.js]: Could duration_type be moved to commonTrading in the common.js file?

File: src/javascript/app/pages/trade/common.js [hunk 2]
```diff
@@ -294,6 +294,14 @@ const commonTrading = (() => {
         d: 5,
     };
 
+    const duration_type = {
```
Comment (binary-hamedanchi) [inline src/javascript/app/pages/trade/common.js]: Can we merge this with duration_order?

File: src/javascript/app/pages/trade/contract.js [hunk 3]
```diff
@@ -75,20 +75,24 @@ const Contract = (() => {
 
         const form_barrier = getFormNameBarrierCategory(form_name);
         form               = form_barrier.form_name;
-        barrier            = barrier_category = form_barrier.barrier_category;
+        if (!form) {
+            return;
+        }
+        barrier = barrier_category = form_barrier.barrier_category;
```
Comment (binary-hamedanchi) [inline src/javascript/app/pages/trade/contract.js]: Does anything break by removing the `barrier_category` and use `barrier` instead?

Comment (negarn) [reply src/javascript/app/pages/trade/contract.js]: no, I think in the past it was just used to show we are comparing category with category in the loop below.

**Pull request merged**